src/LocalTime.php

```diff
@@ -662,17 +663,14 @@ public function jsonSerialize(): string
      */
     public function toISOString(): string
     {
-        if ($this->nano === 0) {
-            if ($this->second === 0) {
-                return sprintf('%02u:%02u', $this->hour, $this->minute);
-            } else {
-                return sprintf('%02u:%02u:%02u', $this->hour, $this->minute, $this->second);
-            }
-        }
-
-        $nanos = rtrim(sprintf('%09u', $this->nano), '0');
+        $hasNanos = $this->nano !== 0;
 
-        return sprintf('%02u:%02u:%02u.%s', $this->hour, $this->minute, $this->second, $nanos);
+        // This code is optimized for high performance
+        return ($this->hour < 10 ? '0' . $this->hour : $this->hour)
+            . ':'
+            . ($this->minute < 10 ? '0' . $this->minute : $this->minute)
+            . ($hasNanos || $this->second !== 0 ? ':' . str_pad((string) $this->second, 2, '0', STR_PAD_LEFT) : '')
```

Comment:
Also, I would put `second` first for better readability:

```suggestion
            . ($this->second !== 0 || $this->nano !== 0 ? ':' . str_pad((string) $this->second, 2, '0', STR_PAD_LEFT) : '')
```

It reads better as hour, minute, second, nano, in this order IMO.

Reply:
Oh also, why don't you use the same trick as for `hour` and `minute`, to avoid an `str_pad()` call?

```suggestion
            . ($this->second !== 0 || $this->nano !== 0 ? ':' . ($this->second < 10 ? '0' . $this->second : $this->second) : '')
```

Reply:
Finally, could you please add a test to `LocalTimeTest` to ensure that hours & minutes are always displayed even where they're `0`? I think these are the only tests missing to cover all edge cases.